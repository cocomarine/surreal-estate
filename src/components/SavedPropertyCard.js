import React from "react";
import PropTypes from "prop-types";
import "../styles/saved-property-card.css";

const SavedPropertyCard = ({ propertyId, title, city, onUnsaveProperty }) => {
  return (
    <div className="saved-property-card-container">
      <div className="saved-property-card">
        {title} in {city}
        <button
          className="remove-button"
          type="submit"
          onClick={() => {
            onUnsaveProperty(propertyId);
          }}
        >
          Remove
        </button>
      </div>
    </div>
  );
};

SavedPropertyCard.propTypes = {
  propertyId: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  onUnsaveProperty: PropTypes.func.isRequired,
};

export default SavedPropertyCard;
