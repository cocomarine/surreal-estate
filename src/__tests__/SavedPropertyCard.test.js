import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import SavedPropertyCard from "../components/SavedPropertyCard";

describe("SavedPropertyCard", () => {
  const validProps = {
    propertyId: "test propertyId",
    title: "test title",
    city: "test city",
    onUnsaveProperty: jest.fn(),
  };

  it("renders correctly", () => {
    const rendered = renderer.create(<SavedPropertyCard {...validProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    render(<SavedPropertyCard {...validProps} />);

    expect(screen.getByText("test title in test city")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveClass("remove-button");
  });

  it("clicking button calls correct function", () => {
    render(<SavedPropertyCard {...validProps} />);

    fireEvent.click(screen.getByRole("button"));
    expect(validProps.onUnsaveProperty).toHaveBeenCalled();
    expect(validProps.onUnsaveProperty).toHaveBeenCalledTimes(1);
    expect(validProps.onUnsaveProperty).toHaveBeenCalledWith(
      validProps.propertyId
    );
  });
});
