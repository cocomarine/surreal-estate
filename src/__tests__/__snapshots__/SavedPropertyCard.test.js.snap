// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedPropertyCard renders correctly 1`] = `
<div
  className="saved-property-card-container"
>
  <div
    className="saved-property-card"
  >
    test title
     in 
    test city
    <button
      className="remove-button"
      onClick={[Function]}
      type="submit"
    >
      Remove
    </button>
  </div>
</div>
`;
